### ISSUE TITLE
SDScaledImageForKey crash (QOS: USER_INTERACTIVE)

### ISSUE THREAD

**Issue by willice9527** [rs/SDWebImage#1450]
Hi:
   Sorry to disturb you.I'm using  SDWebImage in my app ,actually just simple method.In recent several months, we've got thousands of(over 5000) crashes cause by SDScaledImageForKey in SDWebImageCompat.m.  I have checked all the issues exist,not similar one found. Here is a crash log from which you may get some useful information:

Crashed: NSOperationQueue 0x16101dfe0 :: NSOperation 0x15ff9d520 (QOS: USER_INTERACTIVE)
EXC_BAD_ACCESS KERN_INVALID_ADDRESS 0x0000000000000000

Thread : Crashed: NSOperationQueue 0x16101dfe0 :: NSOperation 0x15ff9d520 (QOS: USER_INTERACTIVE)
0  UIKit                          0x186d6f58c _UIBuiltinTraitStorageHash + 136
1  CoreFoundation                 0x181707c40 __CFBasicHashRehash + 1432
2  CoreFoundation                 0x181709034 __CFBasicHashAddValue + 100
3  CoreFoundation                 0x1815b49c4 CFDictionarySetValue + 248
4  UIKit                          0x186d6fd30 _UITraitCollectionCacheForBuiltinStorage + 152
5  UIKit                          0x186d72090 +[UITraitCollection traitCollectionWithDisplayScale:] + 52
6  UIKit                          0x1863d5070 -[UIImage initWithCGImage:scale:orientation:] + 236
7  williceProject                          0x1004fb6e0 SDScaledImageForKey (SDWebImageCompat.m:44)
8  williceProject                          0x1004ff3a8 -[SDWebImageDownloaderOperation scaledImageForKey:image:] (SDWebImageDownloaderOperation.m:361)
9  williceProject                          0x1004ff680 -[SDWebImageDownloaderOperation connectionDidFinishLoading:] (SDWebImageDownloaderOperation.m:386)
10 CFNetwork                      0x181e3e6ec __65-[NSURLConnectionInternal _withConnectionAndDelegate:onlyActive:]_block_invoke + 80
11 CFNetwork                      0x181e3e67c -[NSURLConnectionInternal _withConnectionAndDelegate:onlyActive:] + 200
12 CFNetwork                      0x181e3e7f0 -[NSURLConnectionInternal _withActiveConnectionAndDelegate:] + 56
13 CFNetwork                      0x181cab31c ___ZN27URLConnectionClient_Classic26_delegate_didFinishLoadingEU13block_pointerFvvE_block_invoke + 100
14 CFNetwork                      0x181d98e7c ___ZN27URLConnectionClient_Classic18_withDelegateAsyncEPKcU13block_pointerFvP16_CFURLConnectionPK33CFURLConnectionClientCurrent_VMaxE_block_invoke_2 + 108
15 libdispatch.dylib              0x1811295f0 _dispatch_client_callout + 16
16 libdispatch.dylib              0x18113292c _dispatch_block_invoke + 540
17 CFNetwork                      0x181c97a88 RunloopBlockContext::_invoke_block(void const*, void*) + 36
18 CoreFoundation                 0x1815b90ac CFArrayApplyFunction + 68
19 CFNetwork                      0x181c9796c RunloopBlockContext::perform() + 136
20 CFNetwork                      0x181c9782c MultiplexerSource::perform() + 312
21 CFNetwork                      0x181c97658 MultiplexerSource::_perform(void*) + 68
22 CoreFoundation                 0x18168cefc __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
23 CoreFoundation                 0x18168c990 __CFRunLoopDoSources0 + 540
24 CoreFoundation                 0x18168a690 __CFRunLoopRun + 724
25 CoreFoundation                 0x1815b9680 CFRunLoopRunSpecific + 384
26 CoreFoundation                 0x181606e2c CFRunLoopRun + 112
27 williceProject                          0x1004fde68 -[SDWebImageDownloaderOperation start] (SDWebImageDownloaderOperation.m:117)
28 Foundation                     0x18207e728 __NSOQSchedule_f + 224
29 libdispatch.dylib              0x1811295f0 _dispatch_client_callout + 16
30 libdispatch.dylib              0x181135634 _dispatch_queue_drain + 864
31 libdispatch.dylib              0x18112d0f4 _dispatch_queue_invoke + 464
32 libdispatch.dylib              0x181137504 _dispatch_root_queue_drain + 728
33 libdispatch.dylib              0x181137224 _dispatch_worker_thread3 + 112
34 libsystem_pthread.dylib        0x18133d470 _pthread_wqthread + 1092
35 libsystem_pthread.dylib        0x18133d020 start_wqthread + 4

we use your lib like this:
[self.logoView sd_setImageWithURL:[NSURL URLWithString:_message.iconURL]];

I'll appreciate if you can do some help.Meanwhile it will also make your  SDWebImage more stable.
If you need some more information to fix this issue,please let me know.You can also contact me via email(350775959@qq.com).

**Comment by willice9527** [comment]
anyone for help?

**Comment by bpoplauschi** [comment]
@willice9527 is this still an issue for you or you found a resolution? Which version of the library are you using? Can you reproduce the crash with the demo project?

**Comment by willice9527** [comment]
sorry for response so late. it's nearly impossible to reproduce (the probability of crash for this reason was 0.1% or even less. we got millions of user). we have not do anything to deal with it , cause we haven't a clue about it.
but the good thing is , crashes caused by this seems disappeared recently. now we are using sdwebimage version 3.8.1 from which we have found some new issue ,maybe i'll open an another issue later.
thx
